### ISSUE TITLE
symlink support: readlink() and lstat() do not work 

### ISSUE THREAD

**Issue by drmuey** [CpanelInc/Test-MockFile#25]
Test::MockFile symlinks seem broken, am I doing it wrong?

The object looks right:

```perl -MTest::MockFile -MData::Dumper -e 'my $sym=Test::MockFile->symlink("trg", "/foo/bar");print Dumper($sym)'
$VAR1 = bless( {
                 'nlink' => 0,
                 'gid' => 0,
                 'mode' => 45055,
                 'atime' => 1544466989,
                 'tty' => 0,
                 'rdev' => 0,
                 'fileno' => 900,
                 'inode' => 0,
                 'uid' => 0,
                 'readlink' => 'trg',
                 'mtime' => 1544466989,
                 'blksize' => 4096,
                 'contents' => undef,
                 'dev' => 0,
                 'file_name' => '/foo/bar',
                 'ctime' => 1544466989
               }, 'Test::MockFile' );
```

but then `readlink()` and `lstat()` don’t return what you'd expect:

```
perl -MTest::MockFile -MData::Dumper -e 'my $sym=Test::MockFile->symlink("trg", "/foo/bar");print Dumper({ $sym->{file_name} => readlink($sym->{file_name}) })'
$VAR1 = {
          '/foo/bar' => undef
        };
```

```
perl -MTest::MockFile -MData::Dumper -e 'my $sym=Test::MockFile->symlink("trg", "/foo/bar");print Dumper({ $sym->{file_name} => [lstat($sym->{file_name})] })'
$VAR1 = {
          '/foo/bar' => []
        };
```

`readlink()` should be `trg` and `lstat()` should have much of the data from the object dump?

**Comment by toddr** [comment]
example code please.

**Comment by drmuey** [comment]
> example code please.

already did :)

**Comment by toddr** [comment]
There are 2 feature requests here. 

the lstat issue is resolved by 1c443ae7fa5f4d16b39a00dbda5e4c2155b9fca4

**Comment by toddr** [comment]
readlink support is implemented in 381588dc1463d332a135a3bf2dc137da57bea64c
